app/controllers/authors_controller.rb

```diff
@@ -0,0 +1,52 @@
+class AuthorsController < ApplicationController
+  def show
+    @author = Author.find(params[:id])
+  end
+
+  def new
+    @author = Author.new
+  end
+
+  def create
+    @author = Author.new(author_params)
+    if @author.save
+      redirect_to @author
+    else
+      render 'new'
+    end
+  end
+
+  def edit
+    @author = Author.find(params[:id])
+  end
+
+  def update
+    @author = Author.find(params[:id])
+
+    if @author.update(author_params)
+      redirect_to @author
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    @author = Author.find(params[:id])
+    @author.books.each do |book|
+      book.destroy if book.authors.count == 1
```

Comment:
`count == 1` is quite confusing: why 1?
Would be better this way:
```ruby
book.destroy if book.owned_by?(@author)
```

Then you need to define `owned_by?` method in Book model that returns true only when book has single author, which is given by parameter.